### PR TITLE
Add raffles to xenos

### LINKED ISSUE #2362: Feature request: rafflez 4 xenoes
## Description

After going on da xeno side during the first playtest, any and all xeno spawn were given to whoever could spam the mouse harder on were the request button could be.
Adding raffles would allow to watch the game better, and you know, they just exist so might as well slap them, maybe a raffle of 5 second should be enough

### LINKED ISSUE #2383: Xenos need ghost role raffling (short)

### PR DESCRIPTION
**Pull request by Hmeister-real**
## About the PR
adds raffles. il test later
fixes #2362

## Why / Balance
So people have a good chance to get xenos

## Media

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: Hmeister
- tweak: Xenos now have a ghost role raffle.

### CODE REVIEW

**Comment by UnicornOnLSD** [comment]
also fixes #2362 o7

**Comment by Hmeister-real** [comment]
are raffles not in codebase? shit i expected the raffle system to be here. I guess il port raffles over... later...

**Comment by Tunguso4ka** [comment]
> are raffles not in codebase? shit i expected the raffle system to be here. I guess il port raffles over... later...

зображення
they`re in.

File: Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml [hunk 1]
```diff
@@ -4,6 +4,8 @@
   id: MobXeno
   parent: SimpleSpaceMobBase
   description: They mostly come at night. Mostly.
+  raffle:
+    settings: default
```
Comment (Tunguso4ka) [inline Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml:8]: it shouldn't be here.
raffles are a part of `GhostRole` component.

**Comment by Hmeister-real** [comment]
i do NOT know yaml C# or whatever. Need help on this one

File: Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml (hunk 1 above)
Comment (Hmeister-real) [reply Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml:8]: just checked, theres already a line for raffles at 108-109

**Comment by Tunguso4ka** [comment]
add `raffles` to `- type: GhostRole` component in `Resources/Prototypes/_CM14/Entities/Mobs/Xeno/base_xeno.yml` on the line 168

File: Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml (hunk 1 above)
Comment (Tunguso4ka) [reply Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml]: Oh... and also thats xenos from the upstream.
Our content are located in `_CM14` folders.
Path to our xenos - `Resources/Prototypes/_CM14/Entities/Mobs/Xeno/`

File: Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml
Comment (Tunguso4ka) [inline Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml:1]: you forgor to remove changes from this file.

Comment (Hmeister-real) [reply Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml:1]: oh all of them?
k

File: Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml [hunk 2]
```diff
@@ -320,6 +320,8 @@
   name: Spitter
   parent: MobXeno
   id: MobXenoSpitter
+  raffle:
+    settings: default
```
Comment (Tunguso4ka) [inline Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml:324]: you forgor :trollface:

Comment (Hmeister-real) [reply Resources/Prototypes/Entities/Mobs/NPCs/xeno.yml:324]: RAAAAH

**Review by Tunguso4ka — APPROVED** [review]
Good work!
Sadly i dont have `write` access, so here hold my approve.

**Comment by Hmeister-real** [comment]
This is good to go @DrSmugleaf @QuietlyWhisper

**Review by DrSmugleaf — APPROVED** [review]
Thanks for the PR! And to @Tunguso4ka for the reviews

**Pull request merged**